file.go

```diff
@@ -134,37 +149,20 @@ func ioCompletionProcessor(h syscall.Handle) {
 
 // asyncIo processes the return value from ReadFile or WriteFile, blocking until
 // the operation has actually completed.
-func (f *win32File) asyncIo(c *ioOperation, deadline time.Time, bytes uint32, err error) (int, error) {
+func (f *win32File) asyncIo(c *ioOperation, timeout timeoutChan, bytes uint32, err error) (int, error) {
```

Comment:
asyncIO would need to be made aware of if it's in a read or a write to choose which channel to wait on. Doable, but not sure if it makes things more complicated than needed.